modules/openy_gc_autologout/openy_gc_autologout.module

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Contains openy_gc_autologout.module.
+ */
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+
+/**
+ * Implements hook_page_attachments().
+ */
+function openy_gc_autologout_page_attachments(array &$attachments) {
+  $currentRouteMatch = \Drupal::service('current_route_match');
+  $route_name = $currentRouteMatch->getRouteName();
+  switch ($route_name) {
+    case 'entity.node.canonical':
+      $node = $currentRouteMatch->getParameter('node');
+      $currentUser = \Drupal::service('current_user');
+      if ($currentUser->isAuthenticated() && openy_gc_autologout_check_if_prgf_at_node($node, 'gated_content')) {
+        $autologout_timeout = \Drupal::configFactory()->get('openy_gc_autologout.settings')->get('autologout_timeout');
+        $virtual_y_logout_url = \Drupal::configFactory()->get('openy_gated_content.settings')->get('virtual_y_logout_url');
+        $redirect_url = !empty($virtual_y_logout_url) ? $virtual_y_logout_url : Url::fromRoute('<front>')->toString();
+        $attachments['#attached']['library'][] = 'openy_gc_autologout/openy_gc_autologout';
+        $attachments['#attached']['drupalSettings']['openy_gc_autologout'] = [
+          'autologout_timeout' => $autologout_timeout,
+          'redirect_url' => $redirect_url,
+        ];
+      }
+      break;
+  }
+}
+
+/**
+ * Check if provided paragraph exists on the node.
+ */
+function openy_gc_autologout_check_if_prgf_at_node(NodeInterface $node, $paragraph_id) {
+  $connection = Database::getConnection();
+
+  $result = $connection->select('paragraphs_item_field_data', 'pd')
```

Comment:
I think it is good chance to make a separate service for this code and for existed one:
https://github.com/fivejars/openy_gated_content/blob/master/modules/openy_gc_auth/src/EventSubscriber/VirtualYLoginRedirect.php#L155